### PR TITLE
feat: migrate in_app_messaging to sound null safety

### PR DESCRIPTION
**Pull request by atrope**
## Description

This migrates firebase_in_app_messaging to non-nullable types

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

This is a "Breaking" Change because it is upgrading to null safety, the user code will not be needed to change as the dart lib was almost ready to be null safe already checking for null value and throwing error

### CODE REVIEW

File: packages/firebase_in_app_messaging/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 1.0.0
+
+ - **CHORE**: bump packages.
+ - **BREAKING** **FEAT**: Migrate firebase_in_app_messaging to sound null safety
```
Comment (rrousselGit) [inline packages/firebase_in_app_messaging/CHANGELOG.md:4]: Changelogs are generated, could you remove this?

**Comment by rrousselGit** [comment]
Thanks! Could you update the tests too?

File: packages/firebase_in_app_messaging/CHANGELOG.md (hunk 1 above)
Comment (atrope) [reply packages/firebase_in_app_messaging/CHANGELOG.md]: done

**Comment by atrope** [comment]
@rrousselGit sure! Updated also the example to use Sound Null safety and tested :)

File: packages/firebase_in_app_messaging/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: firebase_in_app_messaging
 description: Flutter plugin for Firebase In-App Messaging.
-version: 0.4.0+1
+version: 1.0.0
```
Comment (rrousselGit) [inline packages/firebase_in_app_messaging/pubspec.yaml:3]:
```suggestion
version: 0.4.0+1
```

This is done at release step

**Review by rrousselGit — APPROVED** [review]
LGTM after you've removed the version bump :smile:

**Comment by atrope** [comment]
> LGTM after you've removed the version bump 😄

Didn't know that it was done only at release.
Done!

**Comment by atrope** [comment]
Hey @rrousselGit ,
What is the scheduled plan for merging and bumping version?
This is the last plugin needed for us to go full sound null safety and maybe some others are also needing this bump

**Comment by rrousselGit** [comment]
We typically do release on Monday. You can use a git dependency in the meantime

**Comment by zetflo** [comment]
Hello, can you merge this PR ?
🙏

**Comment by zetflo** [comment]
👍

**Comment by Salakar** [comment]
This has now been published to pub.dev - thanks everyone.

**Pull request merged**